apphub/semantic_segmentation/unet3d/3DUnetMRBrainSegmentation_Isensee.ipynb

```diff
@@ -0,0 +1,908 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Training UNet3D network for tumor detection in MR images using Brats 2018 data\n",
+    "The notebook is an attempt to train a network to segment tumors from MR brain images. The paper that is being attempted here is \n",
+    "https://arxiv.org/pdf/1802.10508.pdf . The current implementation has been guided by the preprocessing and patching logic from https://github.com/ellisdg/3DUnetCNN for fastestimator environment."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%reload_ext autoreload\n",
+    "%autoreload 2\n",
+    "%matplotlib inline"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from fastestimator.dataset import brats\n",
+    "import pandas as pd\n",
+    "import nibabel as nib\n",
+    "import os\n",
+    "from nilearn import plotting\n",
+    "import numpy as np\n",
+    "import fastestimator as fe\n",
+    "from fastestimator.architecture import UNet3D_Isensee\n",
+    "from fastestimator.op import TensorOp, NumpyOp\n",
+    "from fastestimator.op.numpyop import ImageReader, NIBImageReader\n",
+    "from fastestimator.trace import LRController, ModelSaver\n",
+    "from fastestimator.op.tensorop.augmentation import Augmentation3D\n",
+    "from fastestimator.op.tensorop import Loss, ModelOp\n",
+    "from fastestimator.op.tensorop.loss import BinaryCrossentropy, WeightedDiceLoss\n",
+    "import tensorflow as tf"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Please follow the link to download  https://www.med.upenn.edu/sbia/brats2018/data.html. One has to register at the site.\n",
+    "Having downloaded the Brats 2018 dataset , created folder structure\n",
+    "   <br> ~/fastestimator_data/Brats \n",
+    "   <br> ~/fastestimator_data/BraTs/data\n",
+    "   <br> ~/fastestimator_data/BraTs/data/LGG\n",
+    "   <br> ~/fastestimator_data/BraTs/data/HGG \n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Let visualize the BraTs 2018 data we are dealing with. \n",
+    "Shall be reading nifti image pertaining to  t1 modality as well as the segmentation ground truth using the nibable package. Nilearn plotting are utilized to display mask and image overlay."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "path_brats = '/home/ubuntu/fastestimator_data/BraTs/'"
```

Comment:
can this string also be '~/fastestimator_data/BraTs' or will that fail execution?

Reply:
os.path.join(os.getenv('HOME'),'fastestimator_data/BraTs/') . Tried to avoid /home/ubuntu in the notebook.